### PR TITLE
feat(test): add one-time backup function tests

### PR DESCRIPTION
**Pull request by zhangyifan27**
### What problem does this PR solve?
Add  one-time backup function tests.

##### Tests 

- Fuction test : test backup and restore using `local_service`

##### Code changes

- Has exported new tests

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]

File: src/test/function_test/test_backup_and_restore.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                return false;
+            }
+            // we got only one backup_item for a certain app_id and backup_id.
+            auto item = resp.backup_items[0];
+            is_backup_complete = (item.end_time_ms > 0);
+        }
+        return is_backup_complete;
+    }
+
+    bool wait_app_become_healthy(const std::string &app_name, int max_sleep_seconds)
```
Comment (levy5307) [inline src/test/function_test/test_backup_and_restore.cpp:176]: int --> uint32_t

File: src/test/function_test/test_backup_and_restore.cpp [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    {
+        int sleep_sec = 0;
+        bool is_app_healthy = false;
+        while (!is_app_healthy && sleep_sec <= max_sleep_seconds) {
+            std::cout << "sleep 10s to wait app become healthy." << std::endl;
+            sleep(10);
```
Comment (levy5307) [inline src/test/function_test/test_backup_and_restore.cpp:182]:
```
uint8_t sleep_once_sec = 10;
sleep(sleep_once_sec);
sleep_sec += sleep_once_sec;
```

File: src/test/function_test/test_backup_and_restore.cpp [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            if (value != expected_value) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    start_backup_app_response start_backup(const std::string &user_specified_path = "")
+    {
+        auto err_resp = _ddl_client->backup_app(_old_app_id, _provider, user_specified_path);
```
Comment (levy5307) [inline src/test/function_test/test_backup_and_restore.cpp:133]: return _ddl_client->backup_app(_old_app_id, _provider, user_specified_path).get_value();

File: src/test/function_test/test_backup_and_restore.cpp [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        return err_resp.get_value();
+    }
+
+    query_backup_status_response query_backup(int64_t backup_id)
+    {
+        auto err_resp = _ddl_client->query_backup(_old_app_id, backup_id);
```
Comment (levy5307) [inline src/test/function_test/test_backup_and_restore.cpp:139]: return _ddl_client->query_backup(_old_app_id, backup_id).get_value();

File: src/test/function_test/test_backup_and_restore.cpp [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+private:
+    std::shared_ptr<replication_ddl_client> _ddl_client;
+
+    const int _num_of_rows;
+    const std::string _cluster_name;
+    const std::string _old_app_name;
+    const std::string _new_app_name;
+    const std::string _provider;
+
+    std::string _pegasus_root_dir;
```
Comment (levy5307) [inline src/test/function_test/test_backup_and_restore.cpp:237]: dsn::string_view _pegasus_root_dir;

Comment (zhangyifan27) [reply src/test/function_test/test_backup_and_restore.cpp:239]: `global_env::instance()._pegasus_root` is a `string`, so maybe no need to convert it to `dsn::string_view`.

**Review by hycdong — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]

**Pull request merged**